src/main/java/com/globalitgeeks/examninja/usermanagement/service/UserService.java

```diff
@@ -56,6 +56,20 @@ private boolean isValidPassword(String password) {
     }
 
     // Login user
+    public User login(String email, String password) throws Exception {
+        Optional<User> userOptional = userRepository.findByEmail(email);
+
+        if (userOptional.isPresent()) {
+            User user = userOptional.get();
+            if (user.getPassword().equals(password)) {
+                return user;
+            } else {
+                throw new Exception("Invalid password");
```

Comment:
Use ControllerAdvice to handle exception. It is the recommended way of handling exceptions in a centralised manner.